### ISSUE TITLE
[DocDB][B&R][Packed Columns] Failed to write a batch with 0 operations into RocksDB: Corruption (yb/docdb/docdb_compaction_context.cc:132): Double packed rows: 

### ISSUE THREAD

**Issue by def-** [yugabyte/yugabyte-db#13846]
Jira Link: [DB-3345](https://yugabyte.atlassian.net/browse/DB-3345)
### Description

On my [puppy-food-arm-1](https://portal.dev.yugabyte.com/universes/01559475-2aac-4cda-9ac8-fd971f17b622/nodes) universe with 2.15.3.0-b77 version and these gflags:
![Screenshot 2022-09-02 at 18 35 54](https://user-images.githubusercontent.com/2335377/188199401-75503588-ed94-4c09-8423-0a041b7f2de5.png)
![Screenshot 2022-09-02 at 18 38 13](https://user-images.githubusercontent.com/2335377/188199461-babe73f2-0bf8-49b6-99e7-5e0f1f3ff676.png)
I just got a master FATAL with corruption following a backup with workloads still running:
```
[yugabyte@ip-10-9-82-24 logs]$ cat yb-master.FATAL.details.2022-09-02T16_32_05.pid376445.txt
F20220902 16:32:05 ../../src/yb/tablet/tablet.cc:1348] T 00000000000000000000000000000000 P d2d9ca80ccd441c0b92c6f9d70e7b099: Failed to write a batch with 0 operations into RocksDB: Corruption (yb/docdb/docdb_compaction_context.cc:132): Double packed rows: 4880000001533030303033336536303030303330303038303030303030303030303034613061000021, 48800000015330303030333365363030303033303030383030303030303030303030346130610000212380016C907786112F804A0100000000000000
    @     0xffffb78a9c30  (unknown)
    @     0xffffb78a4864  (unknown)
    @     0xffffb78a519c  (unknown)
    @     0xffffb78a7928  (unknown)
    @     0xffffbb2cc2c4  yb::tablet::Tablet::WriteToRocksDB()
    @     0xffffbb2cb584  yb::tablet::Tablet::ApplyKeyValueRowOperations()
    @     0xffffbb2cb384  yb::tablet::Tablet::ApplyOperation()
    @     0xffffbb2cb018  yb::tablet::Tablet::ApplyRowOperations()
    @     0xffffbb37e528  yb::tablet::WriteOperation::DoReplicated()
    @     0xffffbb35cb48  yb::tablet::Operation::Replicated()
    @     0xffffbb369320  yb::tablet::OperationDriver::ApplyTask()
    @     0xffffbb369100  yb::tablet::OperationDriver::ReplicationFinished()
    @     0xffffba631630  yb::consensus::ConsensusRound::NotifyReplicationFinished()
    @     0xffffba670ce4  yb::consensus::ReplicaState::ApplyPendingOperationsUnlocked()
    @     0xffffba670424  yb::consensus::ReplicaState::AdvanceCommittedOpIdUnlocked()
    @     0xffffba656634  yb::consensus::RaftConsensus::EarlyCommitUnlocked()
    @     0xffffba653bc8  yb::consensus::RaftConsensus::UpdateReplica()
    @     0xffffba652e1c  yb::consensus::RaftConsensus::Update()
    @     0xffffbb71a85c  yb::tserver::ConsensusServiceImpl::UpdateConsensus()
    @     0xffffb91c8e90  std::__1::__function::__func<>::operator()()
    @     0xffffb91c8b24  yb::consensus::ConsensusServiceIf::Handle()
    @     0xffffb815a05c  yb::rpc::ServicePoolImpl::Handle()
    @     0xffffb8107cc0  yb::rpc::InboundCall::InboundCallTask::Run()
    @     0xffffb816506c  yb::rpc::(anonymous namespace)::Worker::Execute()
    @     0xffffb7c85dec  yb::Thread::SuperviseThread()
    @     0xffffb6657898  start_thread
    @     0xffffb65a1ddc  thread_start
```
Backup of YCQL finished at 16:25:
![Screenshot 2022-09-02 at 18 41 22](https://user-images.githubusercontent.com/2335377/188199972-8db6ff31-671c-40f8-87a5-c785b5589cdd.png)
Crashes started at 16:26:
```
-rw-r--r--. 1 yugabyte yugabyte 2.2K Sep  2 16:32 yb-master.FATAL.details.2022-09-02T16_32_05.pid376445.txt
-rw-r--r--. 1 yugabyte yugabyte 1.9K Sep  2 16:31 yb-master.FATAL.details.2022-09-02T16_31_04.pid376268.txt
-rw-r--r--. 1 yugabyte yugabyte 1.9K Sep  2 16:30 yb-master.FATAL.details.2022-09-02T16_30_07.pid375932.txt
-rw-r--r--. 1 yugabyte yugabyte 2.2K Sep  2 16:29 yb-master.FATAL.details.2022-09-02T16_29_04.pid375470.txt
-rw-r--r--. 1 yugabyte yugabyte 2.2K Sep  2 16:28 yb-master.FATAL.details.2022-09-02T16_28_03.pid375335.txt
-rw-r--r--. 1 yugabyte yugabyte 2.2K Sep  2 16:27 yb-master.FATAL.details.2022-09-02T16_27_05.pid375112.txt
-rw-rw-r--. 1 yugabyte yugabyte 1.9K Sep  2 16:26 yb-master.FATAL.details.2022-09-02T16_26_02.pid1357.txt
```
Not sure if we want to consider this as a duplicate of https://github.com/yugabyte/yugabyte-db/issues/12813 or a new bug, depends on if there is a new root cause I guess. Previously I always had this with YSQL, not it seems it happened after the YCQL backup, so might be YCQL related.
I will keep the server in this state for debugging until told otherwise.

**Comment by def-** [comment]
Reproduced by running YCQL backup on second try with the same workloads running and packed columns enabled. So it seems like it doesn’t reproduce 100%, but easy enough.
```
[yugabyte@ip-10-9-118-10 ~]$ cat /home/yugabyte/master/logs/yb-master.FATAL.details.2022-09-12T09_18_02.pid56248.txt
F20220912 09:18:02 ../../src/yb/tablet/tablet.cc:1331] T 00000000000000000000000000000000 P ef2ec92be23b427689d0e80cf007e3a6: Failed to write a batch with 0 operations into RocksDB: Corruption (yb/docdb/docdb_compaction_context.cc:132): Double packed rows: 4880000002533261666561303538663364343436623061663966366430373934363965376563000021, 48800000025332616665613035386633643434366230616639663664303739343639653765630000212380016BC76E694306804A0100000000000000
    @     0xffff872c9c30  (unknown)
    @     0xffff872c4864  (unknown)
    @     0xffff872c519c  (unknown)
    @     0xffff872c7928  (unknown)
    @     0xffff8ad4d6c0  yb::tablet::Tablet::WriteToRocksDB()
    @     0xffff8ad4c980  yb::tablet::Tablet::ApplyKeyValueRowOperations()
    @     0xffff8ad4c780  yb::tablet::Tablet::ApplyOperation()
    @     0xffff8ad4c414  yb::tablet::Tablet::ApplyRowOperations()
    @     0xffff8adfe4c0  yb::tablet::WriteOperation::DoReplicated()
    @     0xffff8adddf8c  yb::tablet::Operation::Replicated()
    @     0xffff8adea080  yb::tablet::OperationDriver::ApplyTask()
    @     0xffff8ade9e60  yb::tablet::OperationDriver::ReplicationFinished()
    @     0xffff8a091b10  yb::consensus::ConsensusRound::NotifyReplicationFinished()
    @     0xffff8a0d19f4  yb::consensus::ReplicaState::ApplyPendingOperationsUnlocked()
    @     0xffff8a0d1134  yb::consensus::ReplicaState::AdvanceCommittedOpIdUnlocked()
    @     0xffff8a0d0ce4  yb::consensus::ReplicaState::UpdateMajorityReplicatedUnlocked()
    @     0xffff8a0b1150  yb::consensus::RaftConsensus::UpdateMajorityReplicated()
    @     0xffff8a087c8c  yb::consensus::PeerMessageQueue::NotifyObserversOfMajorityReplOpChangeTask()
    @     0xffff876b0274  yb::ThreadPool::DispatchThread()
    @     0xffff876a6130  yb::Thread::SuperviseThread()
    @     0xffff86077898  start_thread
    @     0xffff85fc1ddc  thread_start
```

**Comment by def-** [comment]
As suggested by Bogdan, disabling ycql_enable_packed_row on master works around this issue.

**Comment by bmatican** [comment]
Ok, so indeed this seems to be specific to the YCQL packed on master. I suspect all 3 issues you recently filed are in the same bucket.

**Comment by def-** [comment]
The workloads I was running were
```
nm/s1-columns $ips puppy-food-arm-1-s1 --username yugabyte --password $pw
nm/c2 $ips puppy-food-arm-1-c2 --ssl_cert nm/puppy-food-arm-1-ycql.crt --username cassandra --password $pw
nm/s4-columns $ips puppy-food-arm-1-s4 --username yugabyte --password $pw --num_unique_keys 1000000
nm/c4-columns $ips puppy-food-arm-1-c4 --ssl_cert nm/puppy-food-arm-1-ycql.crt --username cassandra --password $pw --num_unique_keys 1000
```
with sample apps 1.8.8 from https://github.com/yugabyte/yb-stress-test/releases
```
[centos@ip ~]$ cat nm/s1-columns
nohup /usr/lib/jvm/java-11-openjdk-11.0.15.0.9-2.el7_9.x86_64/bin/java -jar yb-sample-apps.jar --workload SqlSecondaryIndex --nodes $1  --num_threads_read 1 --num_threads_write 1 --num_unique_keys 1000000000 --num_value_columns 1500 --nouuid "${@:3}" >& $2.log &
[centos@ip ~]$ cat nm/c2
nohup /usr/lib/jvm/java-11-openjdk-11.0.15.0.9-2.el7_9.x86_64/bin/java -jar yb-sample-apps.jar --workload CassandraUniqueSecondaryIndex  --nodes $1  --num_threads_read 1 --num_threads_write 1 --num_unique_keys 1000000000 --nouuid "${@:3}" >& $2.log &
[centos@ip ~]$ cat nm/c4-columns
nohup /usr/lib/jvm/java-11-openjdk-11.0.15.0.9-2.el7_9.x86_64/bin/java -jar yb-sample-apps-1.8.8.jar --workload CassandraDataLoad --nouuid --nodes $1 --num_threads_read 1 --num_threads_write 1 --num_writes -1 --num_reads -1 --num_value_columns 3000 --num_indexes 5 --batch_size 1 --large_value_multiplier 15 --large_key_multiplier 15 --use_datatypes true "${@:3}" >& $2.log &
[centos@ip ~]$ cat nm/s4-columns
nohup /usr/lib/jvm/java-11-openjdk-11.0.15.0.9-2.el7_9.x86_64/bin/java -jar yb-sample-apps-1.8.8.jar --workload SqlDataLoad --use_datatypes true --nodes $1  --num_threads_read 1 --num_threads_write 1 --num_value_columns 1500 --nouuid "${@:3}" >& $2.log &
```

**Comment by bmatican** [comment]
@def- was this consistent? any chance you can re-test with a build that has Sergei's fix above?

**Comment by def-** [comment]
Yes, this one was easy to reproduce. I will retry once our portal.dev.yugabyte.com is on >= b227 and I upgrade my puppy-food-arm-1 LRU to that version.

**Comment by def-** [comment]
Seems to be fixed, couldn't reproduce again on 2.15.4.0-b56
